### ISSUE TITLE
After opening project, click on play timeline icon, play animation twice

### ISSUE THREAD

**Issue by sanjuchopracool** [friction2d/friction#25]
first time it plays animation very fast
second time it plays normally

Expected Behavior:
It should play only once (if loop not selected) with normal speed

Branch : wip (master not tested)

**Comment by rodlie** [comment]
The first run generates the playback cache.

**Comment by sanjuchopracool** [comment]
ohh, I did not look into the code. To a new user it feels like a bug. 
May be we can generate cache without updating canvas in first run ( displaying some message or some user feedback )

**Comment by rodlie** [comment]
To me this is normal behavior and what most (related) programs does. But I'm open for suggestions/discussion.

A better way to visualize cache generation might be enough? Not sure about disabling the canvas during generation, if so it should be an option.

**Comment by sanjuchopracool** [comment]
I am also not good with UX  ideas.  Yes, a better way to visualize cache generation should be enough.

May be just keep this as a low priority issue. Later if others also request, change the priority.

**Comment by rodlie** [comment]
Sure, my current goal it to get the wip branch ready to merge this week(end). Need to fix various ui stuff and try to get the Windows port up-and-running.

**Comment by samytichadou** [comment]
Hi this behavior always bothered me with enve. 
Coming from after effects, the cache generation is at best real time (composition framerate), but often slower

Maybe limits cache generation speed to real time if processing is faster (meaning framerate of the current scene) could be a more intuitive way of dealing with it ?
If it was realtime, perfect, user has seen its timeline, otherwise, cache is ready to play it again.

Cheers !

**Comment by rodlie** [comment]
We could do something about this, but it will require a bit of planning first.

* The simplest solution would be to just wait until enough frames has been buffered, then start the playback (once).
* Another solution would be to buffer frames in the background, giving the user instant playback (in theory)

**Comment by samytichadou** [comment]
> 

I guess after effect does your first suggestion (not sure, a long time since i didn't use it) but i guess both solutions seems a better behavior to me.
Buffer frame in the background sounds perfect, especially for motion graphics, the slow caching would be visible only if scene is too heavy and buffering took to much time. That seems to be more understandable as a user to me. 

Cache visualisation, as said, could also benefits from a new look to make it more understandable for new user !

**Comment by rodlie** [comment]
I will probably have time to do something about this after v0.9.2.

> Cache visualisation, as said, could also benefits from a new look to make it more understandable for new user !

There will be several UI changes.

I'm currently prioritizing features/issues related to work with these first releases. The company I work for use Friction in-house and the priority is to get work done as fast as possible :) (svg animations)

**Comment by samytichadou** [comment]
> I'm currently prioritizing features/issues related to work with these first releases. The company I work for use Friction in-house and the priority is to get work done as fast as possible :) (svg animations)

of course ! not knowing really what is under the hood, the only input i can give is about ux/ui, but having it works rock solid for its primary purpose (svg anim) sound way more important than cache UI :)

**Comment by rodlie** [comment]
I added a progress bar to indicate cache generation, I can't do more until #37 is done.
